### ISSUE TITLE
Get rid of tuples in return values

### ISSUE THREAD

**Issue by MindFlavor** [MindFlavor/AzureSDKForRust#26]
Tuples do not allow you to specify the type in the let binding forcing you to do so in the method call. 
This will lead to non-ergonomic methods if coupled with traits. For example the method ```list_documents``` now returns a tuple:

```rust
fn list_documents<'b, S1, S2, T>(
    &self, 
    database: S1, 
    collection: S2, 
    ldo: &ListDocumentsOptions
) -> impl Future<Item = (ListDocumentsResponse<T>, ListDocumentsResponseAdditionalHeaders), Error = AzureError> 
where
    S1: AsRef<str>,
    S2: AsRef<str>,
    T: DeserializeOwned, 
```
Since there are 2 ```AsRef``` some it would be better if the compiler would be able to infer the type of ```T```. It cannot using a tuple. This is an example of what happens:

```rust
let (entries, ldah) = core.run(
        client.list_documents::<_, _, MySampleStructOwned>(&database_name, &collection_name, &ldo),
    ).unwrap();
```

If it were a single return binding something like this could work instead:

```rust
let response : Response<MySampleStructOwned> = core.run(
        client.list_documents(&database_name, &collection_name, &ldo),
    ).unwrap();
```

With three less useless types specified.

**Comment by MindFlavor** [comment]
Implemented in 59c4497b6698cadc228c4ab5adde126520defac5. Will close when merged in master.

**Comment by MindFlavor** [comment]
Completed in version 0.4.3 ([2e15800d68d8470ea373f1b8b5e5006ce1e861af](2e15800d68d8470ea373f1b8b5e5006ce1e861af)].
